### PR TITLE
IS-15139: added support for paging in the 'project-list' operation

### PR DESCRIPTION
**Pull request by knutj42**

### CODE REVIEW

**Review by ifarshgar — APPROVED** [review]

File: templates/projects.json [hunk 1]
```diff
@@ -1,10 +1,67 @@
...
+        "properties": {
+          "primary_key": "id",
+          "rest_system": "{{@ system @}}",
+          "share_dataset": "{{@ system @}}-{{@ datatype @}}-share"
+        },
+        "template": "transform-collect-rest",
+        "type": "template"
+      },
+      {
+        "rules": {
```
Comment (ellis-igneri) [inline templates/projects.json]: Why does this transform exist?

File: manifest.json [hunk 2]
```diff
@@ -4,15 +4,18 @@
...
     "projects": {
       "template": "templates/projects.json"
     },
     "users": {
-      "template": "templates/projects.json"
+      "template": "templates/users.json"
     },
-    "issues": {
-      "template": "templates/projects.json"
+    "users-youtrack": {
```
Comment (ellis-igneri) [inline manifest.json]: Don't put dashes in datatype names. This should be usersyoutrack. But also, if the set of users returned by usersyoutrack-list is the same as the set returned by users-list AND the set of data contained in usersyoutrack-list exists in the set of data in users-list, then as mentioned in the standup you don't need the usersyoutrack datatype at all.

File: templates/users.json [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        "rules": {
+          "default": [
+            [
+              "copy",
+              "*"
+            ],
+            [
+              "merge",
+              [
+                "apply-hops",
```
Comment (ellis-igneri) [inline templates/users.json]: Never hop in a connector, if there is a need to merge let it happen in the mapping. But as mentioned in the comment in manifest you can probably get away with only collecting the hub users. If there is a reason to collect both let me know and we can talk through the best pattern for that.

**Review by ellis-igneri — CHANGES_REQUESTED** [review]

File: templates/projects.json (hunk 1 above)
Comment (knutj42) [reply templates/projects.json]: just old crud left over from a failed attempt to calculate a "$last-modified" value. I've removed it now.

File: manifest.json (hunk 2 above)
Comment (knutj42) [reply manifest.json]: I've removed the dashes now.

File: templates/users.json (hunk 3 above)
Comment (knutj42) [reply templates/users.json]: I've removed the merging now. We still need both "users" and "usersyoutrack". "usersyoutrack" is basically a mapping between the user-ids used in the youtrack api and the user-ids of the "Hub" users. I've added some documentation about the users here: https://github.com/sesam-io/youtrack-connector/tree/IS-15139#about-users

**Review by ellis-igneri — APPROVED** [review]

**Pull request merged**